src/xrSound/SoundRender_Core.h

```diff
@@ -54,7 +55,7 @@ class CSoundRender_Core : public ISoundManager
     CDB::MODEL* geom_ENV;
 
     // Containers
-    xr_vector<CSoundRender_Source*> s_sources;
+    xr_unordered_map<std::string, CSoundRender_Source*> s_sources;
```

Comment:
Please, don't use `std::` types, since we have our own. At least we should use `xr_string`... But, can you try to use `shared_str` here?